platform-tests/src/test/java/org/junit/platform/launcher/core/LauncherFactoryTests.java

```diff
@@ -390,6 +418,33 @@ static class JUnit5Example {
 		@Test
 		void testJ5() {
 		}
+	}
+
+	@SuppressWarnings({ "JUnitMalformedDeclaration" })
+	@ExtendWith(JupiterExtensionExample.class)
+	static class JupiterTestCase {
+
+		@Test
+		void dummyTest() {
+			// Just a placeholder to trigger the extension
+		}
+	}
 
+	static class LauncherSessionListenerExample implements LauncherSessionListener {
+		@Override
+		public void launcherSessionOpened(LauncherSession session) {
+			session.getStore().put(Namespace.GLOBAL, "testKey", "testValue");
+		}
+	}
+
+	static class JupiterExtensionExample implements BeforeAllCallback {
+		@Override
+		public void beforeAll(ExtensionContext context) {
+			var value = context.getRoot().getStore(ExtensionContext.Namespace.GLOBAL).get("testKey");
```

Comment:
There should be no need to call `getRoot()` here. Please also check that we can get the value from `context.getSessionLevelStore(ExtensionContext.Namespace.GLOBAL)` as well.